### DISCUSSION TITLE
Load assets from external server

### DISCUSSION THREAD

**Discussion by godha89** [stephanrauh/ngx-extended-pdf-viewer#1329] [Q&A]
Hi Team,
In my project, all the assets are fetched from different server than that of index.html.
How can I configure pdf viewer to fetch assets from that external server by giving absolute url?
currently, the assets folder url is created using index.html reference.

**Comment by stephanrauh** [comment]
`pdfDefaultOptions. assetsFolder` allows you to use a custom assets folder. You're right, if you're using a relativ URL, it's relative to the path of the `index.html`. But nothing stops you from using an absolute URL, I guess.

However, you need to solve the CORS issue first. Modern browsers don't allow fetching the assets from a different server, unless that server adds the CORS headers to the response header. That's why I never tested this option.

**Reply by godha89** [reply]
Yes I am able to use absolute url but it breaks with the locale files now.
I can see that the pdf viewer is trying to locate locale file at path like
'Current host'/'custom asset server url'/locale
Here the 'current host' is breaking the loader in my case

**Reply by stephanrauh** [reply]
You can use the attribute `[localeFolderPath]`. The default value is  `'./' + pdfDefaultOptions.assetsFolder + '/locale'`, which matches your error description.

**Reply by godha89** [reply]
It worked loading locales but now failing with pdf.worker.ja files

**Reply by stephanrauh** [reply]
That URL can be modified by overwriting `pdfDefaultOptions.workerSrc`. There are a few more URL defined in the default options:
- sandboxBundleSrc
- standardFontDataUrl
- cMapUrl

I've also found the attribute `[imageResourcesPath]` - but I don't know if there are any files that still use this property.

I suppose that's all paths now!
